### DISCUSSION TITLE
Any recommendations on adding search?

### DISCUSSION THREAD

**Discussion by Svish** [contentlayerdev/contentlayer#444] [Q&A]
I've started porting my old WordPress blog over to MDX-files, with `contentlayer` and `next`. Process is going pretty well, but wondering if there's an easy/good way to add simple search functionality to the website?

Is there a package or library or something that fits well with `contentlayer`, something easy to use and easy to configure, for this purpose?

**Comment by ghost** [comment]
@Svish any solution?

**Reply by Svish** [reply]
I think the Fuse.js library might be a possibility, but as a total newb I need to figure out how to configure it better as things feel a bit "off" now. Like, it works, but it's not as "accurate" as I want it to be, or at least how I felt my old WordPress blog was.

It's good enough for simple searches though, so for now that's what I've used on my website.

**Reply by ghost** [reply]
Thank you for the information 👍.

**Comment by johanguse** [comment]
Hey @skdishansachin , I was looking for the same thing, and I came across someone who implemented a search using Kbar or Algolia. You can find the code reference at https://github.com/timlrx/tailwind-nextjs-starter-blog. I plan to try it in the next couple of weeks.

As @Svish suggested, fuse.js is seen as an option. Here is a sample: https://webcache.googleusercontent.com/search?q=cache:QULWeT8CDxwJ:https://eindm.de/blog/fuzzy-search-nextjs-api-routes&hl=en&gl=br

Let me know if any of these three options work for you!

**Reply by ghost** [reply]
I thought I couldn't use Contentlayer because of search, but you gave it new life. This means a lot to me. I also tried @Svish, worth checking out.
